plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHivePartitionedBucketFunction.java

```diff
@@ -0,0 +1,133 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+import io.trino.spi.connector.BucketFunction;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeOperators;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.trino.block.BlockAssertions.createLongRepeatBlock;
+import static io.trino.block.BlockAssertions.createLongSequenceBlock;
+import static io.trino.block.BlockAssertions.createLongsBlock;
+import static io.trino.plugin.hive.HiveType.HIVE_LONG;
+import static io.trino.plugin.hive.util.HiveBucketing.BucketingVersion;
+import static io.trino.plugin.hive.util.HiveBucketing.BucketingVersion.BUCKETING_V1;
+import static io.trino.plugin.hive.util.HiveBucketing.BucketingVersion.BUCKETING_V2;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static java.util.AbstractMap.SimpleImmutableEntry;
+import static org.testng.Assert.assertEquals;
+
+public class TestHivePartitionedBucketFunction
+{
+    @DataProvider(name = "hiveBucketingVersion")
+    public static Object[][] hiveBucketingVersion()
+    {
+        return new Object[][] {{BUCKETING_V1}, {BUCKETING_V2}};
+    }
+
+    @Test(dataProvider = "hiveBucketingVersion")
+    public void testSinglePartition(BucketingVersion hiveBucketingVersion)
+    {
+        int numValues = 1024;
+        Block bucketColumn = createLongSequenceBlock(923402935, 923402935 + numValues);
+        Page bucketedColumnPage = new Page(bucketColumn);
+        Block partitionColumn = createLongRepeatBlock(78758, numValues);
+        Page page = new Page(bucketColumn, partitionColumn);
+        BucketFunction hiveBucketFunction = bucketFunction(hiveBucketingVersion, 10, ImmutableList.of(HIVE_LONG));
+        Map<Integer, List<Integer>> bucketPositions = new HashMap<>();
+
+        for (int i = 0; i < numValues; i++) {
+            int hiveBucket = hiveBucketFunction.getBucket(bucketedColumnPage, i);
+            // record list of positions for each hive bucket
+            bucketPositions.computeIfAbsent(hiveBucket, ignored -> new ArrayList<>()).add(i);
+        }
+
+        BucketFunction hivePartitionedBucketFunction = partitionedBucketFunction(hiveBucketingVersion, 10, ImmutableList.of(HIVE_LONG), ImmutableList.of(BIGINT), 100);
```

Comment:
please also check that `hivePartitionedBucketFunction` generates 10 distinct values